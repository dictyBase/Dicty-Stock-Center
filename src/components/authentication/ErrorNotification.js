// @flow
import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import SnackbarContent from "@material-ui/core/SnackbarContent"

const useStyles = makeStyles({
  snackbar: {
    backgroundColor: "#cc0000",
  },
})

type Props = {
  /** The error message to display */
  error: string,
}

/** Notification snackbar-style message if user hits some type of error */

const ErrorNotification = ({ error }: Props) => {
  const classes = useStyles()

  return (
    <center>
      <SnackbarContent className={classes.snackbar} message={error} />
      <br />
      <br />
    </center>
  )
}

export default ErrorNotification
