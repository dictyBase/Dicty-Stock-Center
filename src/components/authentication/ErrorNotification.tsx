import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import SnackbarContent from "@material-ui/core/SnackbarContent"

const useStyles = makeStyles({
  snackbar: {
    backgroundColor: "#cc0000",
  },
})

type Props = {
  /** The error message to display */
  error: string
}

/** Notification snackbar-style message if user hits some type of error */

const ErrorNotification = ({ error }: Props) => {
  const classes = useStyles()

  return (
    <div style={{ textAlign: "center" }}>
      <SnackbarContent className={classes.snackbar} message={error} />
      <br />
      <br />
    </div>
  )
}

export default ErrorNotification
