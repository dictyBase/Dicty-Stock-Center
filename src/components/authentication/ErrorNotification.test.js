import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import ErrorNotification from "./ErrorNotification"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<ErrorNotification />, div)
})

test("matching a snapshot of ErrorNotification", () => {
  const component = renderer.create(<ErrorNotification />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
