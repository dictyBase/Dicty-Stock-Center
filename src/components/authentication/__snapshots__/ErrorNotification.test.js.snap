// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ErrorNotification 1`] = `
.c0.c0 {
  background: #cc0000;
}

<center>
  <div
    className="MuiTypography-root-31 MuiTypography-body1-40 MuiPaper-root-4 MuiPaper-elevation6-12 MuiSnackbarContent-root-1 c0"
    role="alertdialog"
  >
    <div
      className="MuiSnackbarContent-message-2"
    />
  </div>
  <br />
  <br />
</center>
`;
